src/handlers/testing/exec/index.ts

```diff
@@ -308,19 +310,21 @@ class RunManager {
     };
     metadata?: unknown;
   }): Promise<void> {
-    let passed: boolean | undefined = undefined;
+    let passed: boolean | null;
+
+    // A pass/fail status is only set if there is a threshold
     if (args.threshold) {
       passed = this.evaluationPassed({
         score: args.score,
         threshold: args.threshold,
       });
+    } else {
+      passed = null;
     }
 
     emitter.emit(EventName.EVALUATION, {
       ...args,
-      // The test case progress component uses null to differentiate between
-      // "not run yet" and "has no pass / fail status"
```

Comment:
I think this comment is still nice to keep

Reply:
i think it made sense to explain why i was flipping `undefined` to `null` but now that i'm just keeping it as `null` there's not really a good place for it. let me actually just switch this to an enum to make it all more clear

Reply:
Ya the reason I mentioned it is because I almost commented to use undefined instead of null then I read the comment that was removed